src/modalities/utils/mfu.py

```diff
@@ -0,0 +1,94 @@
+import warnings
+from typing import Optional, Tuple
+
+import torch
+from torch.distributed.fsdp import FullyShardedDataParallel as FSDP
+from torch.types import Number
+
+from modalities.util import get_total_number_of_trainable_parameters
+
+# A100: https://developer.nvidia.com/blog/nvidia-ampere-architecture-in-depth/
+# H100: https://developer.nvidia.com/blog/nvidia-hopper-architecture-in-depth/
+#       https://www.nvidia.com/en-us/data-center/h100/
+#
+# NOTE: These values are valid for fp16 and bf16 only
+PEAK_PERFORMANCE = {
+    "A100": 312e12,
+    "H100": 989e12,
+}
+
+
+def _get_theoretical_gpu_peak_performance_single(precision: torch.dtype, gpu_type: str) -> Optional[Number]:
+    """
+    returns theoretical gpu peak performance for #GPU=1 in units FLOPs / s for given gpu type
+    """
+    if precision in [torch.float16, torch.bfloat16] and gpu_type in PEAK_PERFORMANCE.keys():
+        return PEAK_PERFORMANCE[gpu_type]
+    else:
+        return None
+
+
+def get_theoretical_gpu_peak_performance(model: FSDP, world_size: int) -> Optional[Number]:
+    """
+    returns theoretical gpu peak performance for #GPU=world_size in units FLOPs / s for given gpu type
+    """
+    if torch.cuda.is_available() and torch.cuda.device_count() > 0:  # necessary for cpu-only tests
+        precision = model.mixed_precision.param_dtype
+        if model.mixed_precision.reduce_dtype != precision or model.mixed_precision.buffer_dtype != precision:
+            warnings.warn("could not get theoretical gpu peak performance for given mixed precision type")
+            return None
```

Comment:
```suggestion
            warnings.warn(f"Could not get theoretical GPU peak performance for given mixed precision type {precision}.")
```

Reply:
I guess you meant to improve the warning message, not to replace `return None`? I fixed this (and related warning messages) in e5af3a44d0f63669b0005083975662f1332e5db1